utils/checks.py

```
@@ -0,0 +1,146 @@
+        must_exists (bool): Should the directory exists?
+        must_be_writable (bool): Should the check test if directory is writable?
+        desc (str): Short description of the value being checked; used in error
+        messages.
+
+    Raises:
+        InvalidConfigurationError: If `path` does not point to a directory or
+        is not writable when required.
+    """
+    if not os.path.exists(path):
+        if must_exists:
+            raise InvalidConfigurationError(f"{desc} '{path}' does not exist")
```

Ruff (0.14.8)

Avoid specifying long messages outside the exception class

(TRY003)